core/src/test/scala/unit/kafka/admin/ConsumerGroupCommandTest.scala

```diff
@@ -0,0 +1,202 @@
+                               group: String = group,
+                               strategy: String = classOf[RangeAssignor].getName): ConsumerGroupExecutor = {
+    val executor = new ConsumerGroupExecutor(brokerList, numConsumers, group, topic, strategy)
+    addExecutor(executor)
+    executor
+  }
+
+  def addSimpleGroupExecutor(partitions: Iterable[TopicPartition] = Seq(new TopicPartition(topic, 0)),
+                             group: String = group): SimpleConsumerGroupExecutor = {
+    val executor = new SimpleConsumerGroupExecutor(brokerList, group, partitions)
+    addExecutor(executor)
+    executor
+  }
+
+  private def addExecutor(executor: AbstractConsumerGroupExecutor): AbstractConsumerGroupExecutor = {
+    consumerGroupExecutors = executor :: consumerGroupExecutors
+    executor
+  }
+
+}
+
+object ConsumerGroupCommandTest {
+
+  abstract class AbstractConsumerThread(broker: String, groupId: String) extends Runnable {
+    val props = new Properties
+    configure(props)
+    val consumer = new KafkaConsumer(props)
+
+    def configure(props: Properties): Unit = {
+      props.put("bootstrap.servers", broker)
+      props.put("group.id", groupId)
+      props.put("key.deserializer", classOf[StringDeserializer].getName)
+      props.put("value.deserializer", classOf[StringDeserializer].getName)
+    }
+
+    def subscribe(): Unit
+
+    def run() {
+      try {
+        subscribe()
+        while (true)
+          consumer.poll(Long.MaxValue)
+      } catch {
+        case _: WakeupException => // OK
+      } finally {
+        consumer.close()
+      }
+    }
+
+    def shutdown() {
+      consumer.wakeup()
+    }
+  }
+
+  class ConsumerThread(broker: String, groupId: String, topic: String, strategy: String)
+    extends AbstractConsumerThread(broker, groupId) {
+
+    override def configure(props: Properties): Unit = {
+      super.configure(props)
+      props.put("partition.assignment.strategy", strategy)
+    }
+
+    override def subscribe(): Unit = {
+      consumer.subscribe(Collections.singleton(topic))
+    }
+  }
+
+  class SimpleConsumerThread(broker: String, groupId: String, partitions: Iterable[TopicPartition])
+    extends AbstractConsumerThread(broker, groupId) {
+
+    override def subscribe(): Unit = {
+      consumer.assign(partitions.toList.asJava)
+    }
+  }
+
+  class AbstractConsumerGroupExecutor(numThreads: Int) {
+    private val executor: ExecutorService = Executors.newFixedThreadPool(numThreads)
+    private val consumers = new ArrayBuffer[AbstractConsumerThread]()
+
+    def submit(consumerThread: AbstractConsumerThread) {
+      consumers += consumerThread
+      executor.submit(consumerThread)
+    }
+
+    def shutdown() {
+      consumers.foreach(_.shutdown())
+      executor.shutdown()
+      try {
+        executor.awaitTermination(5000, TimeUnit.MILLISECONDS)
+      } catch {
+        case e: InterruptedException =>
+          e.printStackTrace()
```

Comment:
Why is `InterruptedException` caught here? Don't we want to wait for termination anyway?